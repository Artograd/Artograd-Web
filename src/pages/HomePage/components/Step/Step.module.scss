.stepWrapper {
  flex: 50%;
}

.stepWrapper:nth-child(even) {
  align-items: flex-end;
}

.step {
  width: 470px;
  margin-bottom: 120px;
}

.offsetStart {
  margin-top: 60px;
}

.offsetEnd {
  margin-bottom: 30px;
}

.stepNumber {
  color: var(--night500);
  font-size: 36px;
  line-height: 40px;
  font-family: var(--font-museo-sans);
  margin: 0;
  font-weight: 600;
}

.stepTitle {
  margin: 0;
  padding-top: 12px;
  color: var(--night950);
  font-size: 18px;
  line-height: 24px;
  font-family: var(--uui-font);
  font-weight: 600;
  padding-bottom: 0;
}

.stepDescription {
  color: var(--night700);
  font-family: var(--uui-font);
}

.stepLink {
  text-transform: capitalize;
}
