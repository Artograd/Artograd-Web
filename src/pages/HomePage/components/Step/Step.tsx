import { FlexRow, FlexCell, Text, Panel, LinkButton } from '@epam/uui';
import styles from './Step.module.scss';

type StepProps = {
  id: number;
  title: string;
  iconOnTop: boolean;
  description: string;
  icon?: string;
  linkUrl?: string;
  linkText?: string;
};

export const Step = ({
  id,
  title,
  iconOnTop,
  icon,
  description,
  linkUrl,
  linkText,
}: StepProps) => {
  return (
    <Panel cx={styles.stepWrapper}>
      <Panel
        cx={`${styles.step} ${id === 2 && styles.offsetStart} ${
          icon && !iconOnTop && styles.offsetEnd
        }`}
      >
        <FlexRow alignItems="bottom">
          <FlexCell width="auto">
            <Text cx={styles.stepNumber}>
              {id.toString().padStart(2, '0')}.
            </Text>
            <Text cx={styles.stepTitle}>{title}</Text>
          </FlexCell>
          {iconOnTop && icon && (
            <FlexCell width="auto">
              <img src={icon} className={styles.stepIcon} />
            </FlexCell>
          )}
        </FlexRow>
        <Text cx={styles.stepDescription} size="48">
          {description}
        </Text>
        {((linkText && linkUrl) || (!iconOnTop && icon)) && (
          <FlexRow alignItems="top">
            {linkText && linkUrl && (
              <FlexCell width="auto">
                <LinkButton
                  caption={linkText}
                  link={{ pathname: linkUrl }}
                  size="42"
                  captionCX={styles.stepLink}
                />
              </FlexCell>
            )}
            {!iconOnTop && icon && (
              <FlexCell width="auto">
                <img src={icon} className={styles.stepIcon} />
              </FlexCell>
            )}
          </FlexRow>
        )}
      </Panel>
    </Panel>
  );
};
