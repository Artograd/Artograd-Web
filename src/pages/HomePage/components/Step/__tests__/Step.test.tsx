import { render } from '@testing-library/react';
import { Step } from '../Step';
import EmptyIcon from '../../../../../images/empty.svg';

const mockData = {
  id: 1,
  title: 'Tenders for art objects are opened',
  iconOnTop: false,
  description:
    'State officials announce open calls for street art tenders, inviting sculptors, artists, and creators to enhance the cityscape with original art installations.',
  linkUrl: '/',
  linkText: 'DISCOVER OPEN TENDERS',
  icon: EmptyIcon,
};

describe('Step component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('renders correctly', () => {
    const component = render(<Step {...mockData} />);

    expect(component).toMatchSnapshot();
  });

  test('renders correctly with offset and icon on top', () => {
    const component = render(<Step {...mockData} id={2} iconOnTop />);

    expect(component).toMatchSnapshot();
  });

  test('renders correctly without the link', () => {
    const component = render(
      <Step {...mockData} linkUrl={undefined} linkText={undefined} />,
    );

    expect(component).toMatchSnapshot();
  });
});
