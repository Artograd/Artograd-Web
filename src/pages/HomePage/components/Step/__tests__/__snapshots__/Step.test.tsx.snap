// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="uui-panel lNErC1 stepWrapper -YmdJu"
      >
        <div
          class="uui-panel lNErC1 step false offsetEnd -YmdJu"
        >
          <div
            class="uui-flex-row z4sbms _1HO2Le x8mfAd _7a2mcl"
          >
            <div
              class=""
              style="min-width: 0; flex-basis: auto;"
            >
              <div
                class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography b2hPSQ -KuFcE ACLoGo stepNumber VocdbV"
              >
                01
              </div>
              <div
                class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography b2hPSQ -KuFcE ACLoGo stepTitle VocdbV"
              >
                Tenders for art objects are opened
              </div>
            </div>
          </div>
          <div
            class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography -Y5Dym _6GiZro _6Zp8gd stepDescription VocdbV"
          >
            State officials announce open calls for street art tenders, inviting sculptors, artists, and creators to enhance the cityscape with original art installations.
          </div>
          <div
            class="uui-flex-row z4sbms _1HO2Le x8mfAd k3cQ11"
          >
            <div
              class=""
              style="min-width: 0; flex-basis: auto;"
            >
              <a
                class="penmb- uui-button-box uui-enabled -clickable uui-link_button tZra-e uui-size-42 uui-no-left-icon uui-has-right-icon uui-color-primary"
                data-testid="step-link-1"
                role="link"
                tabindex="0"
              >
                <div
                  class="uui-caption stepLink"
                >
                  DISCOVER OPEN TENDERS
                </div>
                <div
                  class="_1t9CRQ uui-icon uui-enabled"
                >
                  <svg
                    class=""
                  >
                    navigation-chevron-right-18.svg
                  </svg>
                </div>
              </a>
            </div>
            <div
              class="stepIcon"
              style="min-width: 0; flex-basis: auto;"
            >
              <img
                src="empty.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="uui-panel lNErC1 stepWrapper -YmdJu"
    >
      <div
        class="uui-panel lNErC1 step false offsetEnd -YmdJu"
      >
        <div
          class="uui-flex-row z4sbms _1HO2Le x8mfAd _7a2mcl"
        >
          <div
            class=""
            style="min-width: 0; flex-basis: auto;"
          >
            <div
              class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography b2hPSQ -KuFcE ACLoGo stepNumber VocdbV"
            >
              01
            </div>
            <div
              class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography b2hPSQ -KuFcE ACLoGo stepTitle VocdbV"
            >
              Tenders for art objects are opened
            </div>
          </div>
        </div>
        <div
          class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography -Y5Dym _6GiZro _6Zp8gd stepDescription VocdbV"
        >
          State officials announce open calls for street art tenders, inviting sculptors, artists, and creators to enhance the cityscape with original art installations.
        </div>
        <div
          class="uui-flex-row z4sbms _1HO2Le x8mfAd k3cQ11"
        >
          <div
            class=""
            style="min-width: 0; flex-basis: auto;"
          >
            <a
              class="penmb- uui-button-box uui-enabled -clickable uui-link_button tZra-e uui-size-42 uui-no-left-icon uui-has-right-icon uui-color-primary"
              data-testid="step-link-1"
              role="link"
              tabindex="0"
            >
              <div
                class="uui-caption stepLink"
              >
                DISCOVER OPEN TENDERS
              </div>
              <div
                class="_1t9CRQ uui-icon uui-enabled"
              >
                <svg
                  class=""
                >
                  navigation-chevron-right-18.svg
                </svg>
              </div>
            </a>
          </div>
          <div
            class="stepIcon"
            style="min-width: 0; flex-basis: auto;"
          >
            <img
              src="empty.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Step component renders correctly with offset and icon on top 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="uui-panel lNErC1 stepWrapper -YmdJu"
      >
        <div
          class="uui-panel lNErC1 step offsetStart false -YmdJu"
        >
          <div
            class="uui-flex-row z4sbms _1HO2Le x8mfAd _7a2mcl"
          >
            <div
              class=""
              style="min-width: 0; flex-basis: auto;"
            >
              <div
                class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography b2hPSQ -KuFcE ACLoGo stepNumber VocdbV"
              >
                02
              </div>
              <div
                class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography b2hPSQ -KuFcE ACLoGo stepTitle VocdbV"
              >
                Tenders for art objects are opened
              </div>
            </div>
            <div
              class="stepIcon"
              style="min-width: 0; flex-basis: auto;"
            >
              <img
                src="empty.svg"
              />
            </div>
          </div>
          <div
            class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography -Y5Dym _6GiZro _6Zp8gd stepDescription VocdbV"
          >
            State officials announce open calls for street art tenders, inviting sculptors, artists, and creators to enhance the cityscape with original art installations.
          </div>
          <div
            class="uui-flex-row z4sbms _1HO2Le x8mfAd k3cQ11"
          >
            <div
              class=""
              style="min-width: 0; flex-basis: auto;"
            >
              <a
                class="penmb- uui-button-box uui-enabled -clickable uui-link_button tZra-e uui-size-42 uui-no-left-icon uui-has-right-icon uui-color-primary"
                data-testid="step-link-2"
                role="link"
                tabindex="0"
              >
                <div
                  class="uui-caption stepLink"
                >
                  DISCOVER OPEN TENDERS
                </div>
                <div
                  class="_1t9CRQ uui-icon uui-enabled"
                >
                  <svg
                    class=""
                  >
                    navigation-chevron-right-18.svg
                  </svg>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="uui-panel lNErC1 stepWrapper -YmdJu"
    >
      <div
        class="uui-panel lNErC1 step offsetStart false -YmdJu"
      >
        <div
          class="uui-flex-row z4sbms _1HO2Le x8mfAd _7a2mcl"
        >
          <div
            class=""
            style="min-width: 0; flex-basis: auto;"
          >
            <div
              class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography b2hPSQ -KuFcE ACLoGo stepNumber VocdbV"
            >
              02
            </div>
            <div
              class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography b2hPSQ -KuFcE ACLoGo stepTitle VocdbV"
            >
              Tenders for art objects are opened
            </div>
          </div>
          <div
            class="stepIcon"
            style="min-width: 0; flex-basis: auto;"
          >
            <img
              src="empty.svg"
            />
          </div>
        </div>
        <div
          class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography -Y5Dym _6GiZro _6Zp8gd stepDescription VocdbV"
        >
          State officials announce open calls for street art tenders, inviting sculptors, artists, and creators to enhance the cityscape with original art installations.
        </div>
        <div
          class="uui-flex-row z4sbms _1HO2Le x8mfAd k3cQ11"
        >
          <div
            class=""
            style="min-width: 0; flex-basis: auto;"
          >
            <a
              class="penmb- uui-button-box uui-enabled -clickable uui-link_button tZra-e uui-size-42 uui-no-left-icon uui-has-right-icon uui-color-primary"
              data-testid="step-link-2"
              role="link"
              tabindex="0"
            >
              <div
                class="uui-caption stepLink"
              >
                DISCOVER OPEN TENDERS
              </div>
              <div
                class="_1t9CRQ uui-icon uui-enabled"
              >
                <svg
                  class=""
                >
                  navigation-chevron-right-18.svg
                </svg>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Step component renders correctly without the link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="uui-panel lNErC1 stepWrapper -YmdJu"
      >
        <div
          class="uui-panel lNErC1 step false offsetEnd -YmdJu"
        >
          <div
            class="uui-flex-row z4sbms _1HO2Le x8mfAd _7a2mcl"
          >
            <div
              class=""
              style="min-width: 0; flex-basis: auto;"
            >
              <div
                class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography b2hPSQ -KuFcE ACLoGo stepNumber VocdbV"
              >
                01
              </div>
              <div
                class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography b2hPSQ -KuFcE ACLoGo stepTitle VocdbV"
              >
                Tenders for art objects are opened
              </div>
            </div>
          </div>
          <div
            class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography -Y5Dym _6GiZro _6Zp8gd stepDescription VocdbV"
          >
            State officials announce open calls for street art tenders, inviting sculptors, artists, and creators to enhance the cityscape with original art installations.
          </div>
          <div
            class="uui-flex-row z4sbms _1HO2Le x8mfAd k3cQ11"
          >
            <div
              class="stepIcon"
              style="min-width: 0; flex-basis: auto;"
            >
              <img
                src="empty.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="uui-panel lNErC1 stepWrapper -YmdJu"
    >
      <div
        class="uui-panel lNErC1 step false offsetEnd -YmdJu"
      >
        <div
          class="uui-flex-row z4sbms _1HO2Le x8mfAd _7a2mcl"
        >
          <div
            class=""
            style="min-width: 0; flex-basis: auto;"
          >
            <div
              class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography b2hPSQ -KuFcE ACLoGo stepNumber VocdbV"
            >
              01
            </div>
            <div
              class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography b2hPSQ -KuFcE ACLoGo stepTitle VocdbV"
            >
              Tenders for art objects are opened
            </div>
          </div>
        </div>
        <div
          class="DItj1X uui-text uui-color-primary uui-font-weight-400 uui-font-style-normal uui-typography -Y5Dym _6GiZro _6Zp8gd stepDescription VocdbV"
        >
          State officials announce open calls for street art tenders, inviting sculptors, artists, and creators to enhance the cityscape with original art installations.
        </div>
        <div
          class="uui-flex-row z4sbms _1HO2Le x8mfAd k3cQ11"
        >
          <div
            class="stepIcon"
            style="min-width: 0; flex-basis: auto;"
          >
            <img
              src="empty.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
